qcodes/dataset/measurements.py

```diff
@@ -116,6 +116,8 @@ def add_result(self,
                     input_size = array_size
             elif any(isinstance(value, t) for t in non_array_like_types):
                 pass
+            elif value is None:
```

Comment:
I guess it's still undecided whether we should allow this (but you actually voted against it, @sohailc, by voting for option 1)

Reply:
I will remove this